helm/metrics-server-app/templates/psp.yaml

```diff
@@ -1,3 +1,4 @@
+{{- if le (int .Capabilities.KubeVersion.Minor) 24 }}
```

Comment:
I would prefer that we use https://github.com/prometheus-community/helm-charts/blob/959784a318d1216a486f025a8b69912493bfb2c8/charts/kube-prometheus-stack/templates/prometheus-operator/psp.yaml#L2 instead of version flags